src/main/java/uk/gov/hmcts/ccd/data/casedataaccesscontrol/DefaultRoleAssignmentRepository.java

```diff
@@ -62,6 +62,24 @@ public DefaultRoleAssignmentRepository(final ApplicationParams applicationParams
         this.restTemplate = restTemplate;
     }
 
+    @Override
+    public RoleAssignmentRequestResponse createRoleAssignment(RoleAssignmentRequestResource assignmentRequest) {
+        try {
+            final HttpEntity<Object> requestEntity =
+                new HttpEntity<>(assignmentRequest, securityUtils.authorizationHeaders());
+
+            final var encodedUrl = UriComponentsBuilder.fromHttpUrl(applicationParams.roleAssignmentBaseURL())
+                .build().toUriString();
+            return restTemplate.exchange(new URI(encodedUrl),
+                HttpMethod.POST, requestEntity,
+                RoleAssignmentRequestResponse.class).getBody();
+
+        } catch (Exception e) {
```

Comment:
Resolved in commit 96adc047a20c6d8d70f4b91d4d9b72e58c62b30e

Done, but only to this particular client call.

This change also allowed the removal of the unnecessary URL encoding which is not required for this call as the URL is a fixed known safe constant rather than a URL based on a template and user input.